File: packages/app/src/services/software/definitions/slack.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'slack';
	public name = 'Slack';
	public category = [SoftwareCategory.Social];
	public downloadName = 'SlackSetup.exe';
	public icon = 'slack.png';
	public homepage = 'https://slack.com';

	public async resolveDownloadUrl() {
		return ok('https://slack.com/api/desktop.latestRelease?arch=x64&variant=exe&redirect=true');
	}
}

File: src/lib/software/slack.py
from src.lib.software import BaseSoftware

class Slack(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'SlackSetup.exe'
        self.silent_install_args = ['--silent']

    def resolve_download_url(self):
        self.url_resolved.emit('https://slack.com/api/desktop.latestRelease?arch=x64&variant=exe&redirect=true')
